website/source/docs/vault-integration/index.html.md

```diff
@@ -0,0 +1,351 @@
+
+  1. Create a "nomad-server" policy used by Nomad to create and manage tokens.
+
+  2. Create a Vault token role with the configuration described below.
+
+  3. Configure Nomad to use the created token role.
+
+  4. Give Nomad servers a periodic token with the "nomad-server" policy created
+     above.
+
+#### Required Vault Policies
+
+The token Nomad receives must have the capabilities listed below. An explanation
+for the use of each capability is given.
+
+```hcl
+# Allow creating tokens under "nomad-cluster" token role. The token role name
+# should be updated if "nomad-cluster" is not used.
+path "auth/token/create/nomad-cluster" {
+  capabilities = ["update"]
+}
+
+# Allow looking up "nomad-cluster" token role. The token role name should be
+# updated if "nomad-cluster" is not used.
+path "auth/token/roles/nomad-cluster" {
+  capabilities = ["read"]
+}
+
+# Allow looking up the token passed to Nomad to validate # the token has the
+# proper capabilities. This is provided by the "default" policy.
+path "auth/token/lookup-self" {
+  capabilities = ["read"]
+}
+
+# Allow looking up incoming tokens to validate they have permissions to access
+# the tokens they are requesting. This is only required if
+# `allow_unauthenticated` is set to false.
+path "auth/token/lookup" {
+  capabilities = ["update"]
+}
+
+# Allow revoking tokens that should no longer exist. This allows revoking
+# tokens for dead tasks.
+path "auth/token/revoke-accessor" {
+  capabilities = ["update"]
+}
+
+# Allow checking the capabilities of our own token. This is used to validate the
+# token upon startup.
+path "sys/capabilities-self" {
+  capabilities = ["update"]
+}
+
+# Allow our own token to be renewed.
+path "auth/token/renew-self" {
+  capabilities = ["update"]
+}
+```
+
+The above [`nomad-server` policy](/data/vault/nomad-server-policy.hcl) is
+available for download. Below is an example of writing this policy to Vault:
+
+```
+# Download the policy
+$ curl https://nomadproject.io/data/vault/nomad-server-policy.hcl -O -s -L
+
+# Write the policy to Vault
+$ vault policy write nomad-server nomad-server-policy.hcl
+```
+
+#### Vault Token Role Configuration
+
+A Vault token role must be created for use by Nomad. The token role can be used
+to manage what Vault policies are accessible by jobs submitted to Nomad. The
+policies can be managed as a whitelist by using `allowed_policies` in the token
+role definition or as a blacklist by using `disallowed_policies`.
+
+If using `allowed_policies`, tasks may only request Vault policies that are in
+the list. If `disallowed_policies` is used, task may request any policy that is
+not in the `disallowed_policies` list. There are trade-offs to both approaches
+but generally it is easier to use the blacklist approach and add policies that
+you would not like tasks to have access to into the `disallowed_policies` list.
+
+An example token role definition is given below:
+
+```json
+{
+  "disallowed_policies": "nomad-server",
```

Comment:
Did you really mean to *disallow* `nomad-server` policy, or you meant to specify `allowed_policies`?

Reply:
/guides/operations/vault-integration/index.html